tracer/src/Datadog.Trace/AppSec/EventTrackingSdk.cs

```diff
@@ -17,172 +18,185 @@ namespace Datadog.Trace.AppSec;
+    {
+        TelemetryFactory.Metrics.Record(PublicApiUsage.EventTrackingSdk_TrackUserLoginFailureEvent);
+        TrackUserLoginFailureEvent(userId, exists, null, Tracer.Instance);
+    }
+
+    /// <summary>
+    /// Sets the details of a logon failure on the local root span
+    /// </summary>
+    /// <param name="userId">The userId associated with the login failure</param>
+    /// <param name="exists">If the userId associated with the login failure exists</param>
+    /// <param name="metadata">Metadata associated with the login failure</param>
+    [PublicApi]
+    public static void TrackUserLoginFailureEvent(string userId, bool exists, IDictionary<string, string> metadata)
+    {
+        TelemetryFactory.Metrics.Record(PublicApiUsage.EventTrackingSdk_TrackUserLoginFailureEvent_Metadata);
+        TrackUserLoginFailureEvent(userId, exists, metadata, Tracer.Instance);
+    }
+
+    internal static void TrackUserLoginFailureEvent(string userId, bool exists, IDictionary<string, string> metadata, Tracer tracer)
+    {
+        if (string.IsNullOrEmpty(userId))
         {
-            TelemetryFactory.Metrics.Record(PublicApiUsage.EventTrackingSdk_TrackUserLoginFailureEvent);
-            TrackUserLoginFailureEvent(userId, exists, null, Tracer.Instance);
+            ThrowHelper.ThrowArgumentNullException(nameof(userId));
         }
 
-        /// <summary>
-        /// Sets the details of a logon failure on the local root span
-        /// </summary>
-        /// <param name="userId">The userId associated with the login failure</param>
-        /// <param name="exists">If the userId associated with the login failure exists</param>
-        /// <param name="metadata">Metadata associated with the login failure</param>
-        [PublicApi]
-        public static void TrackUserLoginFailureEvent(string userId, bool exists, IDictionary<string, string> metadata)
-        {
-            TelemetryFactory.Metrics.Record(PublicApiUsage.EventTrackingSdk_TrackUserLoginFailureEvent_Metadata);
-            TrackUserLoginFailureEvent(userId, exists, metadata, Tracer.Instance);
-        }
+        var span = tracer?.ActiveScope?.Span;
 
-        internal static void TrackUserLoginFailureEvent(string userId, bool exists, IDictionary<string, string> metadata, Tracer tracer)
+        if (span is null)
         {
-            if (string.IsNullOrEmpty(userId))
-            {
-                ThrowHelper.ThrowArgumentNullException(nameof(userId));
-            }
-
-            var span = tracer?.ActiveScope?.Span;
-
-            if (span is null)
-            {
-                ThrowHelper.ThrowException("Can't create a tracking event with no active span");
-            }
+            ThrowHelper.ThrowException("Can't create a tracking event with no active span");
+        }
 
-            var setTag = TaggingUtils.GetSpanSetter(span);
+        var setTag = TaggingUtils.GetSpanSetter(span, out var spanInternal);
```

Comment:
Why do we use spanInternal here and not in LoginSuccess?

Reply:
good point, I added it everywhere in [5eb2354](https://github.com/DataDog/dd-trace-dotnet/pull/6225/commits/5eb235421602d965bfe3b2ceb925b15fac34cad7)